django/db/models/fields/related.py

```diff
@@ -731,26 +732,23 @@ def get_reverse_path_info(self):
         pathinfos = [PathInfo(from_opts, opts, (opts.pk,), self.remote_field, not self.unique, False)]
         return pathinfos
 
-    def get_lookup(self, lookup_name):
-        if lookup_name == 'in':
-            return RelatedIn
-        elif lookup_name == 'exact':
-            return RelatedExact
-        elif lookup_name == 'gt':
-            return RelatedGreaterThan
-        elif lookup_name == 'gte':
-            return RelatedGreaterThanOrEqual
-        elif lookup_name == 'lt':
-            return RelatedLessThan
-        elif lookup_name == 'lte':
-            return RelatedLessThanOrEqual
-        elif lookup_name == 'isnull':
-            return RelatedIsNull
-        else:
-            raise TypeError('Related Field got invalid lookup: %s' % lookup_name)
-
-    def get_transform(self, *args, **kwargs):
-        raise NotImplementedError('Relational fields do not support transforms.')
+    def _get_lookup(self, lookup_name):
+        try:
+            return self.class_lookups[lookup_name]
+        except KeyError:
+            # To allow for inheritance, check parent class' class_lookups.
+            for parent in inspect.getmro(self.__class__):
+                # Related lookups need to be explicitly registered to related fields.
+                if parent is ForeignObject:
+                    break
+                if 'class_lookups' not in parent.__dict__:
```

Comment:
> The one concern I have here (I haven't tested it) is that caching will happen on the RegisterLookupMixin

Caching and checking on the `cls.__dict__` should ensure that we don't interact with cached lookups on parent classes (eg, on `RegisterLookupMixin`). Added a test to demonstrate.